src/API/QueryFieldService.php

```diff
@@ -47,49 +48,55 @@ public function __construct(
         $this->queryTypeRegistry = $queryTypeRegistry;
     }
 
-    /**
-     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
-     * @param string $fieldDefinitionIdentifier
-     *
-     * @return \eZ\Publish\API\Repository\Values\Content\Content[]
-     *
-     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
-     */
     public function loadContentItems(Content $content, string $fieldDefinitionIdentifier): iterable
     {
-        $query = $this->prepareQuery($content, $fieldDefinitionIdentifier);
+        $query = $this->prepareQuery($content, $this->locationService->loadLocation($content->contentInfo->mainLocationId), $fieldDefinitionIdentifier);
```

Comment:
You should be able to use 3.0 DX enhancements (proxied & lazy-loaded):
```suggestion
        $query = $this->prepareQuery($content, $content->contentInfo->getMainLocation(), $fieldDefinitionIdentifier);
```